### ISSUE TITLE
CPU

### ISSUE THREAD

**Issue by yongrenr** [kuleshov-group/caduceus#13]
Hello, I want to use cpu to run the code in the trainer module, but it shows that it is not supported. How can I adjust to use the cpu?
Looking forward to your answer!
![image](https://github.com/kuleshov-group/caduceus/assets/78602328/54acec8e-ea44-4de2-adb2-81661a998e5a)

**Comment by yair-schiff** [comment]
I think that this is unfortunately a non-trivial amount of work. Many of the packages assume accelerator is available. I don't think any of the core requirements such as `mamba-ssm`, `flash-attn`, etc are even compatible with CPU.

**Comment by yongrenr** [comment]
Very presumptuous! I have trained one of my models, and its related paths are also placed in outputs. I want to call him directly to test on the test set instead of the validation set. I tried many times and had several problems: 1. The model I trained cannot be used; 2. When using the initial pre-trained model, the validation set is always tested, not the test set.Can you tell me how to use it? Very grateful!
![image](https://github.com/kuleshov-group/caduceus/assets/78602328/bfa808bf-ecf9-4c65-9c55-631b8c57fe93)
![image](https://github.com/kuleshov-group/caduceus/assets/78602328/e6fee41c-f51f-4260-975a-108d579f0eb6)

**Comment by yair-schiff** [comment]
> 1. The model I trained cannot be used;

Can you explain a bit more about what this issue is that is blocking you from using the model?

> 2. When using the initial pre-trained model, the validation set is always tested, not the test set.Can you tell me how to use it? Very grateful!

Set `train.remove_test_loader_in_eval=false`. This should evaluate on both the validation and test set.

**Comment by yongrenr** [comment]
You responded very quickly! I appreciate it!
1.The model I trained cannot be used
Specifically, I used your model to conduct classification experiments. I set up the training and test sets outside the code. I want to train your model on the training set, save it, and then test it separately on the test set. From my query, I think the model is saved under the outputs folder after training. Is this correct?If it is correct, I would like to know how should I fill it in to achieve the above functions?
I used my trained model to test on the test set as I understood it in the picture, but something went wrong.
![image](https://github.com/kuleshov-group/caduceus/assets/78602328/b48f52f3-b018-40ff-b553-cbbd246de934)
![image](https://github.com/kuleshov-group/caduceus/assets/78602328/ccbaec54-eb81-4d85-b657-f84972779906)
![image](https://github.com/kuleshov-group/caduceus/assets/78602328/8022a74d-e61e-4301-af2e-b4c434679d6b)
![8d84e97eb55175a5d1e35d504a014d0](https://github.com/kuleshov-group/caduceus/assets/78602328/d4e8b767-8d73-416f-ab8e-f7151211a57f)
At the same time, I compared the pre-trained model_json with the model_json I saved after running my own training set, and the following error occurred.I want to know how can I make changes? Can you guide me?Really appreciate it!
![image](https://github.com/kuleshov-group/caduceus/assets/78602328/6b51f43b-3753-4a25-bfff-b4b35b3f4a15)

2.Set . This should evaluate on both the validation and test set.train.remove_test_loader_in_eval=false
Sorry, I did not find "train.remove_test_loader_in_eval"，sorry I only found it.Can you tell me where it is set? Or what do I need to do?Really appreciate it!Really, really thank you for your answer!
![image](https://github.com/kuleshov-group/caduceus/assets/78602328/3799910f-1dc3-4000-b6dc-a7bc55be7f49)
